src/utils/config.rs

```diff
@@ -202,6 +212,27 @@ impl ProviderConfig {
             ProviderConfig::TrustedService { .. } => ProviderId::TrustedService,
         }
     }
+    /// Get the name of the Provider
+    /// If there is not one set, use the default.
+    pub fn provider_name(&self) -> String {
+        match *self {
+            ProviderConfig::MbedCrypto { ref name, .. } => name
+                .clone()
+                .unwrap_or_else(|| String::from("mbed-crypto-provider")),
```

Comment:
Could you move the `&str`s to be [associated constants](https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/associated-constants.html) of the providers?